c10/util/string_view.h

```diff
@@ -0,0 +1,677 @@
+#pragma once
+
+#include <c10/macros/Macros.h>
+#include <c10/util/C++17.h>
+#include <c10/util/reverse_iterator.h>
+#include <cstring>
+#include <limits>
+#include <stdexcept>
+#include <string>
+
+namespace c10 {
+
+/**
+ * Reimplementation of std::string_view for C++11.
```

Comment:
Umm... did you implement this yourself?

Reply:
yes, I didn't find a public implementation that fit our needs i.e. is constexpr and works with pre-C++17.